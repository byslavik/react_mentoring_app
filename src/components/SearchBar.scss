.searchBar {
    background: rgba(0,0,0,.5);
    padding: 10px;
    color: #fff;
    margin-top: 10px;
}
.searchBar:after {
    content: '';
    display: block;
    clear: both;
}
.searchBarOptions {
    margin-top: 10px;
    float: left;
}
.searchBarOptions ul {
    display: inline-block;
    padding: 0;
    margin-left: 10px;
}
.searchBarOptions li {
    display: inline-block;
    list-style-type: none;
    margin-right: 10px;

}

.searchBarOptions li a {
    padding: 10px;
}
.searchBar input[type="button"] {
    float: right;
    margin-top: 10px;
}
.searchBarField {
    width: 100%;
    box-sizing: border-box;
    border: 0;
    display: block;
    background: rgba(0,0,0,.6);
    padding: 15px;
    border: 3px solid #ccc;
    color: #fff;
}