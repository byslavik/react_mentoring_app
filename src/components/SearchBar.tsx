import * as React from 'react';
import * as s from './scss/SearchBar.scss';

import {Button} from './Button';
import { connect } from 'react-redux';

import {getFilmsByQuery} from '../actions/films';

import {searchMethod, urlParams} from './../models/common.models'; 
import {ReduxStore} from './../models/redux.model'; 

import * as buttons from '../scss/buttons.scss';
import * as utils from '../scss/utilities.scss';

interface SearchBarState {
  searchWord: string;
  method: searchMethod;
}
interface SearchBarProps {
  match: any;
  getFilmsByQuery: any;
  changeSearchMethod: any;
}

export class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
  constructor(props:SearchBarProps) {
    super(props);

    
    this.state = {
      searchWord: this.props.match.params.query || '',
      method: 'movie'
    }
    
  }
  inputHandler = (el:any) => {
    this.setState({searchWord: el.target.value})
  }
  changeMethod = (method: searchMethod) => {
   this.setState({method: method})
  }
  componentDidMount() {
    if(!this.state.searchWord){
      return;
    }
    let urlParams:urlParams = {
      type: 'search',
      method: this.state.method,
      query: this.state.searchWord
    };
    
    this.props.getFilmsByQuery(urlParams);
  }
  makeSearch = ()=> {
    history.pushState({},'','/search/' + this.state.searchWord); 
    
    let urlParams:urlParams = {
      type: 'search',
      method: this.state.method,
      query: this.state.searchWord
    };
    
    this.props.changeSearchMethod(this.state.method);
    this.props.getFilmsByQuery(urlParams);   
  }
  buttonSearch = (event:any)=> {
    if(event.key == 'Enter'){
      this.makeSearch();
    }
  }
  render() {
      return (<div className={s.searchBar}>
              <h2>Find your movie</h2>
              <input type="text" value={this.state.searchWord} onChange={this.inputHandler} className={s.searchBarField} onKeyPress={this.buttonSearch}/>
              <div className={s.searchBarOptions}>
                  Search By:

                  <ul>
                      <li>
                        <Button isActive={this.state.method == 'movie'} action={this.changeMethod.bind(this,'movie')} >Movie</Button>
                      </li>
                      <li>
                        <Button isActive={this.state.method == 'person'} action={this.changeMethod.bind(this,'person')} >Person</Button>
                      </li>
                  </ul>
              </div>
              <Button action={this.makeSearch} style={buttons.search}>Search</Button>
      </div>);
  }
}
function changeMethod(field:searchMethod) {
  return {
    type: "CHANGE_METHOD",
    payload: field
  }
}
export default connect<any,any,any>(
  (state:ReduxStore) => ({
    method: state.searchMethod
  }),
  (dispatch:any) => ({
    getFilmsByQuery: (urlParams:urlParams)=> {
      dispatch(getFilmsByQuery(urlParams, 'FETCH_FILMS'));
    },
    changeSearchMethod: (method: searchMethod)=>{
      dispatch(changeMethod(method));
    }
   })
)(SearchBar);