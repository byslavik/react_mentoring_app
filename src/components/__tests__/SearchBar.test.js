import * as React from 'react';
import {SearchBar} from '../SearchBar';
import * as renderer from 'react-test-renderer';
import {match} from './testData'

jest.mock('../Button',()=> ({Button:'Button'}));

describe('SearchBar component', ()=>{
  it('SearchBar component shows correctly with "person" method', () => {
    const tree = renderer.create(
      <SearchBar method="person" match={match}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('SearchBar component shows correctly with "movie" method', () => {
    const tree = renderer.create(
      <SearchBar method="movie" match={match}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})