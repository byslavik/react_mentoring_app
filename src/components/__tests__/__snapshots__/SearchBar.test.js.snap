// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component SearchBar component shows correctly with "movie" method 1`] = `
<div
  className={undefined}
>
  <h2>
    Find your movie
  </h2>
  <input
    className={undefined}
    onChange={[Function]}
    onKeyPress={[Function]}
    type="text"
    value=""
  />
  <div
    className={undefined}
  >
    Search By:
    <ul>
      <li>
        <Button
          action={[Function]}
          isActive={true}
        >
          Movie
        </Button>
      </li>
      <li>
        <Button
          action={[Function]}
          isActive={false}
        >
          Person
        </Button>
      </li>
    </ul>
  </div>
  <Button
    action={[Function]}
    style={undefined}
  >
    Search
  </Button>
</div>
`;

exports[`SearchBar component SearchBar component shows correctly with "person" method 1`] = `
<div
  className={undefined}
>
  <h2>
    Find your movie
  </h2>
  <input
    className={undefined}
    onChange={[Function]}
    onKeyPress={[Function]}
    type="text"
    value=""
  />
  <div
    className={undefined}
  >
    Search By:
    <ul>
      <li>
        <Button
          action={[Function]}
          isActive={true}
        >
          Movie
        </Button>
      </li>
      <li>
        <Button
          action={[Function]}
          isActive={false}
        >
          Person
        </Button>
      </li>
    </ul>
  </div>
  <Button
    action={[Function]}
    style={undefined}
  >
    Search
  </Button>
</div>
`;
